packages/node_modules/@ciscospark/spark-core/src/interceptors/rate-limit.js

```diff
@@ -0,0 +1,130 @@
+/*!
+ * Copyright (c) 2015-2017 Cisco Systems, Inc. See LICENSE file.
+ */
+
+import {Interceptor} from '@ciscospark/http-core';
+
+// contains the system time in milliseconds at which the retry after associated with a 429 expires
+// mapped by the API name, e.g.: idbroker.webex.com/doStuff would be mapped as 'doStuff'
+const rateLimitExpiryTime = new WeakMap();
+
+// extracts the common identity API being called
+const idBrokerRegex = /.*(idbroker|identity).webex.com\/([^\/]+)/;
+
+/**
+ * @class
+ */
+export default class RateLimitInterceptor extends Interceptor {
+
+  /**
+   * @returns {RateLimitInterceptor}
+   */
+  static create() {
+    return new RateLimitInterceptor({spark: this});
+  }
+
+  /**
+  * constructor
+  * @param {mixed} args
+  * @returns {Exception}
+  */
+  constructor(...args) {
+    super(...args);
+    rateLimitExpiryTime.set(this, new Map());
+  }
+    /**
+   * @see {@link Interceptor#onRequest}
+   * @param {Object} options
+   * @returns {Object}
+   */
+  onRequest(options) {
+    if (this.isRateLimited(options.uri)) {
+      return Promise.reject(
+        new Error(`API rate limited ${options.uri}`)
+      );
+    }
+    return Promise.resolve(options);
+  }
+  /**
+   * @see {@link Interceptor#onResponseError}
+   * @param {Object} options
+   * @param {Error} reason
+   * @returns {Object}
+   */
+  onResponseError(options, reason) {
+    if (reason.statusCode === 429 && (options.uri.includes(`idbroker`) || options.uri.includes(`identity`))) {
+      // set the retry after in the map, setting to milliseconds
+      this.setRateLimitExpiry(options.uri, this.extractRetryAfterTime(options) * 1000);
```

Comment:
The `*1000` might make more sense inside `extractRetryAfterTime`; that way, all functions' inputs and outputs are in the same unit.

Reply:
I don't really like the idea of the method mutating the form of the data which is used as input, but I don't have strong objections. Consistency is important as well, so I can make that change.